test/v5/kyberBancorReserve.js

```diff
@@ -0,0 +1,576 @@
+        }
+        try {
+            _ = await KyberBancorReserve.new(
+                bancorNetwork.address,
+                network,
+                feeBps.valueOf(),
+                bancorEthToken.address,
+                bancorBntToken.address,
+                zeroAddress
+            );
+            assert(false, "throw was expected in line above.")
+        } catch (e) {
+            assert(Helper.isRevertErrorMessage(e), "expected throw but got: " + e);
+        }
+        _ = await KyberBancorReserve.new(
+            bancorNetwork.address,
+            network,
+            feeBps.valueOf(),
+            bancorEthToken.address,
+            bancorBntToken.address,
+            admin
+        );
+    });
+
+    it("Should test can not set fee of 10000", async function() {
+        await reserve.setFeeBps(feeBps, {from: admin});
+        try {
+            await reserve.setFeeBps(10000, {from: admin});
+            assert(false, "throw was expected in line above.")
+        } catch (e) {
+            assert(Helper.isRevertErrorMessage(e), "expected throw but got: " + e);
+        }
+    });
+
+    it("Should test can not set fee sender is not admin", async function() {
+        await reserve.setFeeBps(feeBps, {from: admin});
+        try {
+            await reserve.setFeeBps(feeBps, {from: user});
+            assert(false, "throw was expected in line above.")
+        } catch (e) {
+            assert(Helper.isRevertErrorMessage(e), "expected throw but got: " + e);
+        }
+    });
+
+    // test setting contracts
+    it("Should test can not set contracts with invalid network or bancor network", async function() {
+        try {
+            await reserve.setKyberNetwork(zeroAddress);
+            assert(false, "throw was expected in line above.")
+        } catch (e) {
+            assert(Helper.isRevertErrorMessage(e), "expected throw but got: " + e);
+        }
+        try {
+            await reserve.setBancorContract(zeroAddress);
+            assert(false, "throw was expected in line above.")
+        } catch (e) {
+            assert(Helper.isRevertErrorMessage(e), "expected throw but got: " + e);
+        }
+        try {
+            await reserve.setKyberNetwork(network, {from: user});
+            assert(false, "throw was expected in line above.")
+        } catch (e) {
+            assert(Helper.isRevertErrorMessage(e), "expected throw but got: " + e);
+        }
+        try {
+            await reserve.setBancorContract(bancorNetwork.address, {from: user});
+            assert(false, "throw was expected in line above.")
+        } catch (e) {
+            assert(Helper.isRevertErrorMessage(e), "expected throw but got: " + e);
+        }
+
+        await reserve.setKyberNetwork(network, {from: admin});
+        await reserve.setBancorContract(bancorNetwork.address, {from: admin});
+    });
+
+    it("Should test trade with trade is not enable", async function() {
```

Comment:
trade reverted when trade is not enabled.

Reply:
Updated